superset/utils/decorators.py

```diff
@@ -29,3 +35,51 @@ def stats_timing(stats_key, stats_logger):
         raise e
     finally:
         stats_logger.timing(stats_key, now_as_float() - start_ts)
+
+
+def etag_cache(max_age, check_perms=bool):
+    """
+    A decorator for caching views and handling etag conditional requests.
+
+    The decorator caches the response, returning headers for etag and last
+    modified. If the client makes a request that matches, the server will
+    return a "304 Not Mofified" status.
+
+    """
+    def decorator(f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            try:
+                check_perms(request)
+                # build the cache key from the function arguments and any other
+                # additional GET arguments (like `form_data`, eg).
+                key_args = list(args)
+                key_kwargs = kwargs.copy()
+                key_kwargs.update(request.args)
+                cache_key = wrapper.make_cache_key(f, *key_args, **key_kwargs)
+                response = cache.get(cache_key)
+            except Exception:  # pylint: disable=broad-except
+                logging.exception('Exception possibly due to cache backend.')
```

Comment:
any chance you can tighten up this exception handling and messaging? Do we really need to disable the linter here? If we handle distinct exception types could we get logging that is more definitive than "possibly due to...."? Sorry to be a stickler, but caching issues can be super frustrating, and this sort of thing can go a long way to short-circuiting debugging work.